assessments/models.py

```
@@ -1,45 +1,62 @@
+    status = models.CharField(
+        max_length=1, choices=SubmissionStatus.choices, default=SubmissionStatus.SUBMITTED, db_index=True
+    )
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(
+                fields=["assignment", "student"],
+                condition=~models.Q(status="i"),  # Only enforce uniqueness for active records
+                name="unique_active_assignment_submission",
+            )
+        ]
```

Ruff (0.15.18)

[warning] Mutable default value for class attribute

(RUF012)